constellation_utils.py

```diff
@@ -2,36 +2,114 @@
 from skyfield.api import load, wgs84, EarthSatellite
 import pandas as pd
 import numpy as np
+import pydeck as pdk
+import plotly.figure_factory as ff
+import plotly.express as px
 
 # SETUP LIST OF ALL CONSTELLATIONS
 # http://systemarchitect.mit.edu/docs/delportillo18b.pdf 
 _MINELEVATIONS = {'SPIRE': 80, 
                   'PLANET': 80, 
                   'STARLINK': 80, 
-                  'SWARM': 80, 
-                  'ONEWEB': 80, 
-                  'GALILEO': 80, 
-                  'BEIDOU': 80, 
-                  'GNSS': 80,
-                  'NOAA': 80, 
+                  'SWARM': 80,
+                  'ONEWEB': 85, 
+                  'GALILEO': 87, 
+                  'BEIDOU': 87, 
+                  'GNSS': 87,
+                  'NOAA': 87, 
                   'IRIDIUM': 80} 
 
+_MAXALTITUDES = {'SPIRE': 600, 
```

Comment:
I'd suggest instead of having separate lists for different elements for each constellation to consolidate those into a list of "constellation" objects, each of which will have the min elevation, max altitude, etc. properties.

Additionally / alternatively, min elevation and max altitude are values you could extract from the raw data (+ an extra buffer if needed). This has the obvious advantage to not require any updates if/when new constellations are added or if a new orbits are added.

Reply:
- Moved constellation specific configs into a different file, and dict format 
- the min elevation is a constraint on the plotting for transits for a given location to limit number of events
- the max altitude comment makes sense, but will add in next iteration